pyCGM_Single/pycgmKinetics.py

```diff
@@ -19,40 +19,300 @@
 #coordinate. Also in Utilities but need to clean everything up somewhat!         
 def f(p, x):
+    """
+    Parameters
+    ----------
+    p : array
+    x : int or float
+        Scaling factor for the first variable in p.
+    Returns
+    -------
+    int or float
+        Returns the first value in p scaled by x, aded by the second value in p.
+
+    Example
+    -------
+    >>> from .pycgmKinetics import *
+    >>> import numpy as np
+    
+    >>> p = [1, 2]
+    >>> x = 10
+    >>> f(p, x) #doctest: +NORMALIZE_WHITESPACE
+    12
+
+    >>> p = np.array([5.16312215, 8.79307163])
+    >>> x = 2.0
+    >>> f(p, x) #doctest: +NORMALIZE_WHITESPACE
+    19.11931593
+    """
     return (p[0] * x) + p[1]
 
 def dot(v,w):
+    """Calculate the Dot Product function.
+    
+    Parameters
+    ----------
+    v : array
+        First 3D vector.
+    w : array
+        Second 3D vector.
+
+    Returns
+    -------
+    int or float
+        The quotient of the dot product of vectors v and w.
+
+    Example
+    -------
+    >>> from .pycgmKinetics import *
+    >>> import numpy as np
+    
+    >>> v = [1, 2, 3]
+    >>> w = [4, 5, 6]
+    >>> dot(v,w) #doctest: +NORMALIZE_WHITESPACE
+    32
+
+    >>> v = np.array([6.56643344, 6.23972959, 2.83918231])
+    >>> w = np.array([5.53732499, 7.61560881, 6.2563037])
+    >>> dot(v,w) #doctest: +NORMALIZE_WHITESPACE
+    101.6426024111329
+    """
     x,y,z = v
     X,Y,Z = w
     return x*X + y*Y + z*Z
   
 def length(v):
+    """Calculate Length of a 3D Vector function.
+    
+    Parameters
+    ----------
+    v : array
+        A 3D vector.
+
+    Returns
+    -------
+    float
+        Returns the length of a 3D vector.
+
+    Example
+    -------
+    >>> from .pycgmKinetics import *
+    >>> import numpy as np
+    
+    >>> v = [1,2,3]
+    >>> length(v) #doctest: +NORMALIZE_WHITESPACE
+    3.7416573867739413
+
+    >>> v = np.array([6.56643344, 6.23972959, 2.83918231])
+    >>> length(v) #doctest: +NORMALIZE_WHITESPACE
+    9.49279883215107
+    """
     x,y,z = v
     return np.sqrt(x*x + y*y + z*z)
   
 def vector(b,e):
+    """Vector Subtraction function.
+    
+    Parameters
+    ----------
+    v : array
+        First 3D vector.
+    e : array
+        Second 3D vector.
+
+    Returns
+    -------
+    tuple
+        Returns the vector of e - v.
+
+    Example
+    -------
+    >>> from .pycgmKinetics import *
+    >>> import numpy as np
+    
+    >>> v = [1,2,3]
+    >>> e = [4,5,6]
+    >>> vector(v, e) #doctest: +NORMALIZE_WHITESPACE
+    (3, 3, 3)
+
+    >>> v = np.array([5.10897693, 6.18161923, 9.44221215])
+    >>> e = np.array([3.68040209, 9.92542233, 5.38362424])
+    >>> vector(v, e) #doctest: +NORMALIZE_WHITESPACE
+    (-1.42857484, 3.7438031, -4.05858791)
+    """
     x,y,z = b
     X,Y,Z = e
     return (X-x, Y-y, Z-z)
   
 def unit(v):
+    """Create Unit Vector function.
+    
+    Parameters
+    ----------
+    v : array
```

Comment:
Please make sure this is accurate, and then specify. For example, is this a python list, or really a numpy array.  If it requres a 1x3 ndarray than specify it as that, or a 3-element list, etc.